distributed/shuffle/shuffle.py

```diff
@@ -70,49 +70,91 @@ def rearrange_by_column_p2p(
             )  # TODO: we fail at non-string object dtypes
     empty[column] = empty[column].astype("int64")  # TODO: this shouldn't be necesssary
 
-    transferred = df.map_partitions(
-        shuffle_transfer,
-        id=token,
-        npartitions=npartitions,
-        column=column,
-        meta=df,
-        enforce_metadata=False,
-        transform_divisions=False,
-    )
```

Comment:
The one downside of this is that we'll no longer get input task fusion. The previous code here was a BlockWise layer and we'll loose this. However, I consider consistency more important than the input task fusion.

FWIW "fixing"/enabling input (and output) task fusion for graphs of this type would be beneficial for other types of graph as well but I consider this out of scope for now